### DISCUSSION TITLE
Default Basemap terria v8

### DISCUSSION THREAD

**Discussion by anujdivesh** [TerriaJS/terriajs#5841] [Q&A] [answered]
Hi, Is there a way to change the default base-map in terria v8. also, where could i get the list of base-map Id's?

Thanks in advance

**Comment by tephenavies** [comment]
Hi @anujdivesh. The default base map is configured in initialization files. See [the section on basemaps configuration](https://docs.terria.io/guide/customizing/initialization-files/#basemaps) in our developer docs. There's also a link from there to the list of default basemap & their ids.

**Comment by anujdivesh** [comment]
Hi Steve, Perfect. While trying to configure the base-map, it does not seem to be reflecting. Do we have to rebuild anything?
For instance:
  "enabledBaseMaps": ["basemap-darkmatter","basemap-positron","basemap-natural-earth-II"],
  "defaultBaseMapId": "basemap-natural-earth-II",

**Comment by zoran995** [comment, marked as answer]
Hi @anujdivesh,
Here is a code snippet you posted above that I just tested, you can also test it here http://ci.terria.io/main/#https://gist.githubusercontent.com/zoran995/186b6da410d4ea38b933eaa489b2d58b/raw/844290860f71dd9f17025ee461c11e162b0409a8/gistfile1.json.
```json5
{
  ... other init config
  "baseMaps": {
    "enabledBaseMaps": ["basemap-darkmatter","basemap-positron","basemap-natural-earth-II"],
    "defaultBaseMapId": "basemap-natural-earth-II"
  },
  ...
}
```
Regarding default-basemap to see the effect of the above configuration you might need to delete a `... .basemap` key from the local storage or use incognito mode in the browser. The order of resolving the base maps is 
- local storage
- init configuration
- default value.

Value in local storage will be saved every time you change the base map from the `Map settings` panel.

**Comment by anujdivesh** [comment]
Hi Thanks so much. It works.
